src/metabase/query_processor/middleware/add_query_throttle.clj

```diff
@@ -0,0 +1,48 @@
+(ns metabase.query-processor.middleware.add-query-throttle
+  "Middleware that constraints the number of concurrent queries, rejects queries by throwing an exception and
+  returning a 503 when we exceed our capacity"
+  (:require [metabase.config :as config]
+            [puppetlabs.i18n.core :refer [tru]])
+  (:import [java.util.concurrent Semaphore TimeUnit]))
+
+(defn- calculate-max-queries-from-max-threads []
```

Comment:
Semantic nitpicking, but why is this a defn, rather than a def?

Reply:
Good catch, I'll fix that